packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -6568,55 +5669,67 @@ export default class Meeting extends StatelessWebexPlugin {
    * @memberof Meeting
    */
   private requestScreenShareFloor() {
-    const content = this.locusInfo.mediaShares.find((element) => element.name === CONTENT);
+    if (!this.mediaProperties.shareTrack || !this.mediaProperties.mediaDirection.sendShare) {
+      LoggerProxy.logger.log(
+        `Meeting:index#requestScreenShareFloor --> NOT requesting floor, because we don't have the share track anymore (shareTrack=${
+          this.mediaProperties.shareTrack ? 'yes' : 'no'
+        }, sendShare=${this.mediaProperties.mediaDirection.sendShare})`
+      );
 
-    if (content && this.shareStatus !== SHARE_STATUS.LOCAL_SHARE_ACTIVE) {
-      Metrics.postEvent({event: eventType.SHARE_INITIATED, meeting: this});
+      return Promise.resolve({});
+    }
+    if (this.state === MEETING_STATE.STATES.JOINED) {
+      const content = this.locusInfo.mediaShares.find((element) => element.name === CONTENT);
 
-      return this.meetingRequest
-        .changeMeetingFloor({
-          disposition: FLOOR_ACTION.GRANTED,
-          personUrl: this.locusInfo.self.url,
-          deviceUrl: this.deviceUrl,
-          uri: content.url,
-          resourceUrl: this.resourceUrl,
-        })
-        .then(() => {
-          this.isSharing = true;
+      if (content && this.shareStatus !== SHARE_STATUS.LOCAL_SHARE_ACTIVE) {
+        Metrics.postEvent({event: eventType.SHARE_INITIATED, meeting: this});
 
-          return Promise.resolve();
-        })
-        .catch((error) => {
-          LoggerProxy.logger.error('Meeting:index#share --> Error ', error);
+        return this.meetingRequest
+          .changeMeetingFloor({
+            disposition: FLOOR_ACTION.GRANTED,
+            personUrl: this.locusInfo.self.url,
+            deviceUrl: this.deviceUrl,
+            uri: content.url,
+            resourceUrl: this.resourceUrl,
+          })
+          .then(() => {
+            this.isSharing = true;
 
-          Metrics.sendBehavioralMetric(BEHAVIORAL_METRICS.MEETING_SHARE_FAILURE, {
-            correlation_id: this.correlationId,
-            locus_id: this.locusUrl.split('/').pop(),
-            reason: error.message,
-            stack: error.stack,
+            return Promise.resolve();
+          })
+          .catch((error) => {
+            LoggerProxy.logger.error('Meeting:index#share --> Error ', error);
+
+            Metrics.sendBehavioralMetric(BEHAVIORAL_METRICS.MEETING_SHARE_FAILURE, {
+              correlation_id: this.correlationId,
+              locus_id: this.locusUrl.split('/').pop(),
+              reason: error.message,
+              stack: error.stack,
+            });
+
+            return Promise.reject(error);
           });
+      }
 
-          return Promise.reject(error);
-        });
+      return Promise.reject(new ParameterError('Cannot share without content.'));
     }
+    this.floorGrantPending = true;
```

Comment:
question: Is this meant to set the floorGrantPending outside of the if JOINED?

Reply:
yes, the logic hasn't changed - it covers the case when we don't receive the Locus DTO with meeting state JOINED before someone calls addMedia with a screen share track. Previously the check `if (this.state === MEETING_STATE.STATES.JOINED)` and setting of `this.floorGrantPending = true;` was done outside the call to `requestScreenShareFloor()` in just 1 place, I moved it inside `requestScreenShareFloor()` so that the check is always done.